compiler/optimizer/VPConstraint.cpp

```diff
@@ -1275,6 +1275,15 @@ TR::VPClassType *TR::VPClassType::create(OMR::ValuePropagation *vp, const char *
    return TR::VPUnresolvedClass::create(vp, sig, len, method);
    }
 
+static bool canBaseClassBeTrusted(TR_OpaqueClassBlock *baseClass)
+   {
+#ifdef J9_PROJECT_SPECIFIC
+   if (TR::Compiler->om.areFlattenableValueTypesEnabled() && TR::Compiler->cls.isValueTypeClass(baseClass))
+      return false;
+ #endif
```

Comment:
As I mentioned in an off-line discussion, I was wondering whether it would be possible to implement this method so that it returns true in OMR, while allowing a downstream project like OpenJ9 to provide its own implementation that might take into account semantics that are particular to that project.  That would avoid introducing new code that needs to be protected by `J9_PROJECT_SPECIFIC`.

Perhaps it could be a method in `OMR::ClassEnv`?

Reply:
> Perhaps it could be a method in OMR::ClassEnv?

I have started the update after our off-line discussion yesterday. Since the purpose of these two methods are similar to `ValuePropagation::isUnreliableSignatureType`, I'm putting these two utility methods into VP as  `isUnreliableSignatureType`.

Reply:
Just running some more tests and should be able to push another commit soon